mitodl/bootcamp-ecommerce
# Adds deal syncing

#### What are the relevant tickets?
#304 #305 

#### What's this PR do?
When smapply application gets updated and approved and payment is made, this updates the corresponding deal in hubspot CRM.

The data mapping to hubspot objects:
Profiles (demographics) -> Contacts
Bootcamps -> Products
PersonalPrice (application stage / price) -> Deals
PersonalPrice (link bootcamp to deal) -> Line Items

#### How should this be manually tested?

1. Create a test account at https://applymitbootcamps.smapply.io/
2. Check hubspot test account for new contact at https://app.hubspot.com/contacts/6431386/contacts/list/view/all/? 
3. Complete Round 1 of the Application, which includes filling out demographic information
4. Check the hubspot test account to see that the contact has been updated
5. Ask an admin to push the your test application through the application process
6. Check the corresponding deal in hubspot to see that it’s been updated
7. Once your application has been approved, make a test payment
8. Confirm that the deal is updated with the payment details
